lib/cretonne/meta/isa/intel/encodings.py

```diff
@@ -169,6 +169,15 @@ def enc_i32_i64_ld_st(inst, w_bit, recipe, *args, **kwargs):
     X86_64.enc(inst.i32.any, *r.rc.rex(0xd3, rrr=rrr))
     X86_64.enc(inst.i32.any, *r.rc(0xd3, rrr=rrr))
 
+for inst,           rrr in [
+        (base.ishl_imm, 4),
+        (base.ushr_imm, 5),
+        (base.sshr_imm, 7)]:
+    X86_32.enc(inst.i32, *r.rib(0xc1, rrr=rrr))
+    X86_64.enc(inst.i64, *r.rib.rex(0xc1, rrr=rrr, w=1))
+    X86_64.enc(inst.i32, *r.rib.rex(0xc1, rrr=rrr))
+    X86_64.enc(inst.i32, *r.rib(0xc1, rrr=rrr))
```

Comment:
Thats correct. I was fooled because you cant use that helper above, because the instructions require the `.any` suffix. (I added a comment there to note why)